### PR TITLE
don't share cookies in private mode

### PR DESCRIPTION
**Pull request by keis**
Work In Progress

For private mode to really work the active cookie jar needs to be cleared as well. That could either be done by a ON_SET trigger or builtin into core. I'm leaning towards having it happen as part of updating the variable as it's feels like a expected part of private browsing and not an optional extra.

and if you actually want that behaviour you could always call load_cookies in a ON_SET

### CODE REVIEW

**Comment by keis** [comment]
rebased on current master.


@mathstuf any input on how to clear the cookie jar?

**Comment by mathstuf** [comment]
Soup has a call for it. Doesn't `cookie clear` exist? Or is that different based on WebKit1 or 2?

**Comment by keis** [comment]
Sorry. I think I was a bit unclear the question regard how we want to behave when changing the private flag.

Scenario 1
User sets `enable_private 1` and only the webkit settings is toggled. the default config has a bit to make uzbl-browser do a `cookie clear` command when enable_private changes

Scenario 2
User sets `enable_private 1` and currently loaded cookies are ignored for further request by core itself. (and possibly recovered if later set to 0 again)

**Comment by mathstuf** [comment]
I think if users want to load cookies back when private browsing is turned off, they can rerun the scripts to load the cookie store. I don't think WebKit has a way to intercept which cookies are sent.

**Comment by keis** [comment]
I was thinking similar to how we clear the cookies we just swap out the cookie jar and then swap it back.

but anyway, do you think cookies should be cleared by turning private on or do we leave that to the config?

**Comment by mathstuf** [comment]
Hmm. Yeah, I think the core should take care of that; too easy to forget otherwise. Private mode has some effect in other places already around the permission handler stuff.

**Comment by keis** [comment]
rebased. @mathstuf I think this should be good to merge if it looks OK to you.

File: src/variables.c [hunk 1]
```diff
@@ -2081,6 +2081,14 @@ IMPLEMENT_SETTER (int, enable_private)
         g_unsetenv (priv_envvar);
     }
 
+    /* Replace the current cookie jar with a new empty jar. */
```
Comment (mathstuf) [inline src/variables.c]: We should only do this if the value has changed.

**Comment by keis** [comment]
Updated. While testing a noticed you get a glib warning if trying to set it to something besides 0 or 1 but it still takes effect and anything >1 is converted to 1. I'm going to ignore that for the "has changed" check as I'm not sure it's something that should be allowed at all.

**Comment by mathstuf** [comment]
I suppose we could actually enforce `gboolean` values in the source code, but that'd be a more general change.

**Comment by keis** [comment]
IMHO it should be done across the board and added as a type to variables.{c,h} and also enforced at that layer. I think it's out of scope for this fix and more of a 1.0+ change

**Pull request merged**